### PR TITLE
feat: add currencyList public query

### PR DESCRIPTION
**Pull request by dolcalmi**
- Add currency list query
- Update price service

https://github.com/GaloyMoney/price/pull/34 is required before merge

How to test:
- Run `make start-deps`
- Run `make start-main-fast`
- Open http://localhost:4002/graphql or use postman to run:
```graphql
query CurrencyList {
  currencyList {
    code
    symbol
    name
    flag
  }
}
```

### CODE REVIEW

File: src/domain/price/index.types.d.ts [hunk 1]
```diff
@@ -10,12 +10,20 @@ type Tick = {
   readonly price: DisplayCurrencyPerSat
 }
 
+type PriceCurrency = {
```
Comment (nicolasburtey) [inline src/domain/price/index.types.d.ts:13]: would be handy to have some description of what this means, maybe with an example.
not clear to me what is the difference between `code` and `symbol`. also what are we going to store in flag? emoji?

Comment (dolcalmi) [reply src/domain/price/index.types.d.ts:13]: we will store the emoji, aslso the code is the ISO code (3 letters) and symbol the currency symbol 

https://github.com/GaloyMoney/price/blob/074e7ca3890a00d94e7acb28089d6c98a132d1b4/realtime/default.yaml#L5

Comment (dolcalmi) [reply src/domain/price/index.types.d.ts:13]: Comments added https://github.com/GaloyMoney/galoy/pull/2068/commits/dc5aa7c48351e03d62127cdf7cbe1adf5293f32b

**Comment by daviroo** [comment]
@dolcalmi this looks good to me.

File: src/app/prices/list-currencies.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import { LocalCacheService } from "@services/cache"
+import { SECS_PER_10_MINS } from "@config"
+
+export const listCurrencies = async (): Promise<PriceCurrency[] | ApplicationError> => {
+  return LocalCacheService().getOrSet({
+    key: CacheKeys.PriceCurrencies,
+    ttlSecs: SECS_PER_10_MINS,
+    getForCaching: () => {
+      return PriceService().listCurrencies()
+    },
```
Comment (nicolasburtey) [inline src/app/prices/list-currencies.ts:12]:
```suggestion
    getForCaching: PriceService().listCurrencies
```

**Review by nicolasburtey — APPROVED** [review]

**Pull request merged**